src/webview/mod.rs

```diff
@@ -135,7 +153,7 @@ pub struct WebViewAttributes {
   /// [bug]: https://bugs.webkit.org/show_bug.cgi?id=229034
   pub custom_protocols: Vec<(
     String,
-    Box<dyn Fn(&Request<Vec<u8>>) -> Result<Response<Cow<'static, [u8]>>>>,
+    Box<dyn Fn(Request<Vec<u8>>, Response) -> Result<()>>,
```

Comment:
It doesn't make sense to return `Result` here if you want to be async. `Response` should be the one to determine the `Result`.

Reply:
pushed @wusyong

Reply:
I mean the implementation should also handle this.
If it's responder dealing with it, it should decide which finish method should be called.

Reply:
We don't necessarily need to add `.respond_with_err()` variant, since you can just create a response with status code between 400-599 and you should be fine. However, I do agree that this function shouldn't return a result to allow the responder to respond even if this function errored and the responder were sent somewhere else.

Reply:
yeah at least it's less confusing for the user where you can call respond() and return an error later and it would crash the app.